### DISCUSSION TITLE
What is the desirable forgot-password flow?

### DISCUSSION THREAD

**Discussion by Tur8008** [fastapi-users/fastapi-users#1263] [Q&A] [answered]
Hello, I just want to be sure I'm implementing the forgot password flow properly as I'm not experienced in auth stuff. So thaks for any recommendations.

If I understood properly It goes like that:
1. User passes down forgot-password page. Fills in an email field and submit. 
2. They get an email with the link to restore-password  page. The link must contain a url to restore password page and the token. So as the developer I must include the token gotten as the param of `on_after_forgot_password` in email message.
3. The must be like `https://domain.name/restore-password?token_contains` - so it is get request
4. I fetch the token from get query param and include it in payload along with new password value.
5. Send post query to `/auth/restore-password/` url and then can inform the user their password has changed using `on_after_reset_password` event handler.

Is that ok?

**Comment by frankie567** [comment, marked as answer]
Yes, that's exactly it!
